Xamarin.Forms.Core/PromptArguments.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Xamarin.Forms.Internals
+{
+	public class PromptArguments
+	{
+		public PromptArguments(string title, string message, string accept, string cancel, string placeholder = null, int? maxLength = null, Keyboard keyboard = default(Keyboard))
+		{
+			if (string.IsNullOrWhiteSpace(accept) || string.IsNullOrWhiteSpace(cancel))
+				throw new Exception($"You must provide a value for {nameof(accept)} and {nameof(cancel)}.");
+
+			Title = title;
+			Message = message;
+			Accept = accept;
+			Cancel = cancel;
+			Placeholder = placeholder;
+			MaxLength = maxLength;
+			Keyboard = keyboard ?? Keyboard.Default;
+			Result = new TaskCompletionSource<string>();
+		}
+
+		public string Title { get; }
+
+		public string Message { get; }
+
+		public string Accept { get; }
+
+		public string Cancel { get; }
+
+		public string Placeholder { get; }
+
+		public int? MaxLength { get; }
+
+		public Keyboard Keyboard { get; }
+
+		public TaskCompletionSource<string> Result { get; }
```

Comment:
the TCS shouldn't be public.

Reply:
The TCS is public in `AlertArguments`, and it's used in `Page.cs` when we return a task result. How do you want me to handle this?

Reply:
you could expose only the returned Task, not the TCS, but as this is readonly, it's fine